app/Lib/Tools/BackgroundJobsTool.php

```diff
@@ -0,0 +1,647 @@
+     * Restarts workers with status != RUNNING
+     *
+     * @param boolean $waitForRestart
+     * @return void
+     */
+    public function restartDeadWorkers(bool $waitForRestart = false): void
+    {
+        $this->Supervisor->startProcessGroup(self::MISP_WORKERS_PROCESS_GROUP, $waitForRestart);
+    }
+
+    /**
+     * Purge queue
+     *
+     * @param string $queue
+     * @return void
+     */
+    public function purgeQueue(string $queue): void
+    {
+        $this->validateQueue($queue);
+
+        $this->RedisConnection->del($queue);
+    }
+
+    /**
+     * Return Background Jobs status
+     *
+     * @return integer
+     */
+    public function getStatus(): int
+    {
+        $redisStatus = false;
+        $supervisorStatus = false;
+
+        if (!$this->settings['enabled']) {
+            return self::STATUS_NOT_ENABLED;
+        }
+
+        try {
+            $redisStatus = $this->RedisConnection->ping();
+        } catch (Exception $exception) {
+            CakeLog::error("SimpleBackgroundJobs Redis error: {$exception->getMessage()}");
+        }
+
+        try {
+            $supervisorStatus = ($this->Supervisor->getState()['statecode'] === \Supervisor\Supervisor::RUNNING);
+        } catch (Exception $exception) {
+            CakeLog::error("SimpleBackgroundJobs Supervisor error: {$exception->getMessage()}");
+        }
+
+        if ($redisStatus && $supervisorStatus) {
+            return self::STATUS_RUNNING;
+        } elseif (!$redisStatus && !$supervisorStatus) {
+            return self::STATUS_REDIS_AND_SUPERVISOR_NOT_OK;
+        } elseif ($redisStatus && !$supervisorStatus) {
+            return self::STATUS_SUPERVISOR_NOT_OK;
+        } else {
+            return self::STATUS_REDIS_NOT_OK;
+        }
+    }
+
+    /**
+     * Validate queue
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateQueue(string $queue): bool
+    {
+        if (!in_array($queue, self::VALID_QUEUES)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid background job queue, must be one of: [%s]',
+                    implode(', ', self::VALID_QUEUES)
+                )
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate command
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateCommand(string $command): bool
+    {
+        if (!in_array($command, self::ALLOWED_COMMANDS)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Invalid command, must be one of: [%s]',
+                    implode(', ', self::ALLOWED_COMMANDS)
+                )
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * Validate worker name
+     *
+     * @return boolean
+     * @throws InvalidArgumentException
+     */
+    private function validateWorkerName(string $name): bool
```

Comment:
This function receives as name the name of the queue, whilst expecting [queue]_[process_id] when starting a new worker via the UI (called by startWorker()). 

This leads to a notice error on the list() call on 539 and finally an exception thrown on 544.